psutil/_psutil_bsd.c

```diff
@@ -391,6 +399,132 @@ psutil_proc_cmdline(PyObject *self, PyObject *args) {
 }
 
 
+/*
+ * Return process environment as a Python dictionary
+ */
+PyObject *
+psutil_proc_environ(PyObject *self, PyObject *args) {
+    int i, cnt = -1;
+    long pid;
+    char *s, **envs, errbuf[_POSIX2_LINE_MAX];
+    PyObject *py_err=NULL, *py_value=NULL, *py_retdict=NULL;
+    kvm_t *kd;
+#ifdef PSUTIL_NETBSD
+    struct kinfo_proc2 *p;
+#else
+    struct kinfo_proc *p;
+#endif
+
+    if (!PyArg_ParseTuple(args, "l", &pid))
+        return NULL;
+
+#if defined(PSUTIL_FREEBSD)
+    kd = kvm_openfiles(NULL, "/dev/null", NULL, 0, errbuf);
+#else
+    kd = kvm_openfiles(NULL, NULL, NULL, KVM_NO_FILES, errbuf);
+#endif
+    if (!kd) {
+        convert_kvm_err("kvm_openfiles", errbuf);
+        return NULL;
+    }
+
+    py_retdict = PyDict_New();
+    if (!py_retdict)
+        goto error;
+
+#if defined(PSUTIL_FREEBSD)
+    p = kvm_getprocs(kd, KERN_PROC_PID, pid, &cnt);
+#elif defined(PSUTIL_OPENBSD)
+    p = kvm_getprocs(kd, KERN_PROC_PID, pid, sizeof(*p), &cnt);
+#elif defined(PSUTIL_NETBSD)
+    p = kvm_getproc2(kd, KERN_PROC_PID, pid, sizeof(*p), &cnt);
+#endif
+    if (!p) {
+        py_err = NoSuchProcess("kvm_getprocs");
+        goto error;
+    }
+    if (cnt <= 0) {
+        py_err = NoSuchProcess(cnt < 0 ? kvm_geterr(kd) : "kvm_getprocs: cnt==0");
+        goto error;
+    }
+
+    /* On *BSD kernels there are a few kernel-only system processes without an environment.
+     * (See e.g. "procstat -e 0 | 1 | 2 ..." on FreeBSD.)
+     *
+     * Some system process have no stats attached at all (they are marked with P_SYSTEM.)
+     *
+     * On FreeBSD, it's possible that the process is swapped or paged out,
+     * then we have no access to the environment stored in the process' user area.
+     *
+     * On NetBSD, we cannot call kvm_getenvv2() for a zombie process.
+     *
+     * To make unittest suite happy, return an empty environment.
+     */
```

Comment:
Just a minor style complaint: could you please use `//` comments and limit the length of the string to 80 chars?